vote/src/vote_state_view.rs

```diff
@@ -0,0 +1,378 @@
+    }
+
+    pub fn commission(&self) -> u8 {
+        let offset = self.frame.offset(Field::Commission);
+        self.data[offset]
+    }
+
+    pub fn votes_iter(&self) -> impl Iterator<Item = Lockout> + '_ {
+        self.votes_view().votes_iter()
+    }
+
+    pub fn last_lockout(&self) -> Option<Lockout> {
+        self.votes_view().last_lockout()
+    }
+
+    pub fn last_voted_slot(&self) -> Option<Slot> {
+        self.last_lockout().map(|v| v.slot())
+    }
+
+    pub fn root_slot(&self) -> Option<Slot> {
+        self.root_slot_view().root_slot()
+    }
+
+    pub fn get_authorized_voter(&self, epoch: Epoch) -> Option<&Pubkey> {
+        self.authorized_voters_view().get_authorized_voter(epoch)
+    }
+
+    pub fn num_epoch_credits(&self) -> usize {
+        self.epoch_credits_view().len()
+    }
+
+    pub fn epoch_credits_iter(&self) -> impl Iterator<Item = &EpochCreditsItem> + '_ {
+        self.epoch_credits_view().epoch_credits_iter()
+    }
+
+    pub fn credits(&self) -> u64 {
+        self.epoch_credits_view().credits()
+    }
+
+    pub fn last_timestamp(&self) -> BlockTimestamp {
+        let offset = self.frame.offset(Field::LastTimestamp);
+        let buffer = &self.data[offset..];
+        let mut cursor = std::io::Cursor::new(buffer);
+        BlockTimestamp {
+            slot: solana_serialize_utils::cursor::read_u64(&mut cursor).unwrap(),
+            timestamp: solana_serialize_utils::cursor::read_i64(&mut cursor).unwrap(),
+        }
+    }
+
+    fn votes_view(&self) -> ListView<VotesListFrame> {
+        let offset = self.frame.offset(Field::Votes);
+        ListView::new(self.frame.votes_frame(), &self.data[offset..])
+    }
+
+    fn root_slot_view(&self) -> RootSlotView {
+        let offset = self.frame.offset(Field::RootSlot);
+        RootSlotView::new(self.frame.root_slot_frame(), &self.data[offset..])
+    }
+
+    fn authorized_voters_view(&self) -> ListView<AuthorizedVotersListFrame> {
+        let offset = self.frame.offset(Field::AuthorizedVoters);
+        ListView::new(self.frame.authorized_voters_frame(), &self.data[offset..])
+    }
+
+    fn epoch_credits_view(&self) -> ListView<EpochCreditsListFrame> {
+        let offset = self.frame.offset(Field::EpochCredits);
+        ListView::new(self.frame.epoch_credits_frame(), &self.data[offset..])
+    }
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
+enum VoteStateFrame {
+    V1_14_11(VoteStateFrameV1_14_11),
+    V3(VoteStateFrameV3),
+}
+
+impl VoteStateFrame {
+    /// Parse a serialized vote state and verify structure.
+    fn try_new(bytes: &[u8]) -> Result<Self> {
+        let version = {
+            let mut cursor = std::io::Cursor::new(bytes);
+            solana_serialize_utils::cursor::read_u32(&mut cursor)
+                .map_err(|_err| VoteStateViewError::ParseError)?
+        };
+
+        Ok(match version {
+            0 => return Err(VoteStateViewError::ParseError),
```

Comment:
is there a specific reason to keep this separate from the catch all?

Reply:
ah i see now from the old code - this used to be a supported version and is no longer.

Reply:
nbd because of the _ arm, but the 0 arm doesn't seem to have test coverage